postcss-selector-parser.d.ts

```diff
@@ -311,6 +351,51 @@ declare namespace parser {
          * The attribute name after having been qualified with a namespace.
          */
         readonly qualifiedAttribute: string;
+
+        /**
+         * The case insensitivity flag or an empty string depending on whether this
+         * attribute is case insensitive.
+         */
+        readonly insensitiveFlag : 'i' | '';
+
+        /**
+         * Returns the attribute's value quoted such that it would be legal to use
+         * in the value of a css file. The original value's quotation setting
+         * used for stringification is left unchanged. See `setValue(value, options)`
+         * if you want to control the quote settings of a new value for the attribute or
+         * `set quoteMark(mark)` if you want to change the quote settings of the current
+         * value.
+         *
+         * You can also change the quotation used for the current value by setting quoteMark.
+         **/
+        quoteValue(options?: SmartQuoteMarkOptions): string;
+
+        /**
+         * Set the unescaped value with the specified quotation options. The value
+         * provided must not include any wrapping quote marks -- those quotes will
+         * be interpreted as part of the value and escaped accordingly.
+         * @param value
+         */
+        setValue(value: string, options?: SmartQuoteMarkOptions): void;
+
+        /**
+         * Intelligently select a quoteMark value based on the value's contents. If
+         * the value is a legal CSS ident, it will not be quoted. Otherwise a quote
+         * mark will be picked that minimizes the number of escapes.
+         *
+         * If there's no clear winner, the quote mark from these options is used,
+         * then the source quote mark (this is inverted if `preferCurrentQuoteMark` is
+         * true). If the quoteMark is unspecified, a double quote is used.
+         **/
+        smartQuoteMark(options: PreferredQuoteMarkOptions): QuoteMark;
```

Comment:
Option: this method can be merged with `preferredQuoteMark()` to simply be `getQuoteMark()` if the options hash is optional. If no options hash is provided, it will intelligently quote as needed. If an options hash is provided, it will use the preferred settings.

Reply:
You're describing the behavior of a private function that I have called `_determineQuoteMark(options)` which calls one of these other two depending on the value of the `smart` option. The name `getQuoteMark(options)` seems confusing to me because there's also a property `quoteMark`. But if no (or empty) options are passed to `getQuoteMark()`, it always returns the value of `quoteMark` so maybe it's fine.